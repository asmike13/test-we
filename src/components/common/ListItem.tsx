import { ReactElement } from "react"
import { ListItem, ListItemIcon } from "@mui/material"

interface ICustomListItem {
	icon?: ReactElement
	text?: string
	children?: ReactElement
}

const CustomListItem = ({ icon, text, children }: ICustomListItem) => {
	return (
		<>
			<ListItem disablePadding sx={{ position: 'relative', minHeight: '50px', paddingLeft: '15px' }}>
				{icon && <ListItemIcon>{icon}</ListItemIcon>}
				{text || children}
			</ListItem>
		</>
	)
}

export default CustomListItem