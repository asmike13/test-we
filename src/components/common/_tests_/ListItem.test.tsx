import renderer from 'react-test-renderer';
import ListItem from '../ListItem'
import HomeIcon from '@mui/icons-material/Home';

const props = {
	icon: <HomeIcon />,
	children: <div>toto</div>,
}

it('should render without props as snapshot', async () => {
	const snapshot = renderer.create(<ListItem />)

	expect(snapshot).toMatchSnapshot()
})

it('should render with children props as snapshot', async () => {
	const snapshot = renderer.create(<ListItem {...props} />)

	expect(snapshot).toMatchSnapshot()
})

it('should render with text props as snapshot', async () => {
	const snapshot = renderer.create(<ListItem {...props} text="text" />)

	expect(snapshot).toMatchSnapshot()
})
