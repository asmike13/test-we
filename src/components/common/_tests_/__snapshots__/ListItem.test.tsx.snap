// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with children props as snapshot 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters css-1xtwiqb-MuiListItem-root"
  disabled={false}
>
  <div
    className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="HomeIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
      />
    </svg>
  </div>
  <div>
    toto
  </div>
</li>
`;

exports[`should render with text props as snapshot 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters css-1xtwiqb-MuiListItem-root"
  disabled={false}
>
  <div
    className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="HomeIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
      />
    </svg>
  </div>
  text
</li>
`;

exports[`should render without props as snapshot 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters css-1xtwiqb-MuiListItem-root"
  disabled={false}
/>
`;
